numba/np/polynomial/polynomial_core.py

```diff
@@ -0,0 +1,218 @@
+from numba.extending import (models, register_model, type_callable,
+                             unbox, NativeValue, make_attribute_wrapper, box,
+                             lower_builtin)
+from numba.core import types, cgutils
+from numpy.polynomial.polynomial import Polynomial
+from contextlib import ExitStack
+import numpy as np
+from llvmlite import ir
+from numba.np import numpy_support
+
+
+@register_model(types.PolynomialType)
+class PolynomialModel(models.StructModel):
+    def __init__(self, dmm, fe_type):
+        members = [
+            ('coef', fe_type.coef),
+            ('domain', fe_type.domain),
+            ('window', fe_type.window)
+            # Introduced in NumPy 1.24, maybe leave it out for now
+            # ('symbol', types.string)
+        ]
+        super(PolynomialModel, self).__init__(dmm, fe_type, members)
+
+
+@type_callable(Polynomial)
+def type_polynomial(context):
+    def typer(coef, domain=None, window=None):
+        default_domain = types.Array(types.int64, 1, 'C')
+        default_window = types.Array(types.int64, 1, 'C')
+        default_coef = types.Array(types.double, 1, 'C')
+
+        if isinstance(coef, types.Array) and \
+                all([a is None for a in (domain, window)]):
+            if coef.ndim == 1:
+                return types.PolynomialType(default_coef,
+                                            default_domain,
+                                            default_window)
```

Comment:
Hi @guilhermeleobas, I have used `default_coef` here since the given input for `coef` must have integer dtype, but we want to always store `coef` as Array of doubles. Please let me know if I have a mistake.

Reply:
oh right, `coef` is always an array of doubles.